hgapp/crafting/models.py

```diff
@@ -58,6 +58,18 @@ def __str__(self):
 
     def get_exp_cost_per_artifact(self):
         return self.relevant_power.get_gift_cost() + 1
+    
+    # Calculate cost if this is an upgrade
+    def get_exp_cost_for_crafting_instance(self, artifact):
+        
+        artifact_cost = self.get_exp_cost_per_artifact()
+
+        last_existing_copy = artifact.power_set.filter(parent_power_id=self.relevant_power_full_id).last()
```

Comment:
I'd prefer to use an explicit ordering here (probably creation time), as SQL does not default to any ordering, and thus this may not return what we want depending on the db.